Unable to restore workspace

Reproduce with clean state:

1) In the Home Workspace, open a new Panel
2) Close the App
3) Re-open the App

Results in an exception and no panels.

```
00:00:04.086 [ERROR] Exception while restoring Workspace 7d5a2b2d-c18d-47e6-91dc-27eb21eb9038|System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceViewModel.OptionalPanelOrFirst(Optional`1 optionalPanelId) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs:line 383
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceViewModel.OpenPageInNewTab(PageData pageData, NewTab newTab) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs:line 361
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceViewModel.<>c__DisplayClass50_0.<OpenPage>b__1(NewTab newTab) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs:line 328
   at OneOf.OneOfBase`3.Switch(Action`1 f0, Action`1 f1, Action`1 f2)
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceViewModel.OpenPage(Optional`1 optionalPageData, OpenPageBehavior behavior, Boolean selectTab) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs:line 326
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceController.OpenPage(WorkspaceId workspaceId, Optional`1 pageData, OpenPageBehavior behavior, Boolean selectTab) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:line 321
   at NexusMods.App.UI.WorkspaceSystem.PanelViewModel.AddDefaultTab() in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelViewModel.cs:line 166
   at NexusMods.App.UI.WorkspaceSystem.PanelViewModel.FromData(PanelData data) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelViewModel.cs:line 248
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceViewModel.<>c__DisplayClass60_0.<FromData>b__0(ISourceUpdater`2 updater) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs:line 503
   at DynamicData.Cache.Internal.ReaderWriter`2.DoUpdate(Action`1 updateAction, Action`1 previewHandler, Boolean collectChanges) in /_/src/DynamicData/Cache/Internal/ReaderWriter.cs:line 157
   at DynamicData.Cache.Internal.ReaderWriter`2.Write(Action`1 updateAction, Action`1 previewHandler, Boolean collectChanges) in /_/src/DynamicData/Cache/Internal/ReaderWriter.cs:line 114
   at DynamicData.ObservableCache`2.UpdateFromSource(Action`1 updateAction) in /_/src/DynamicData/Cache/ObservableCache.cs:line 226
   at DynamicData.SourceCache`2.Edit(Action`1 updateAction) in /_/src/DynamicData/Cache/SourceCache.cs:line 55
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceViewModel.FromData(WorkspaceData data) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs:line 477
   at NexusMods.App.UI.WorkspaceSystem.WorkspaceController.FromData(WindowData data) in /ssd-pool/projects/NexusMods/app/main/src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:line 94
```

--- a/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs
+++ b/src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs
@@ -500,8 +500,8 @@ public class WorkspaceViewModel : AViewModel<IWorkspaceViewModel>, IWorkspaceVie
                     WorkspaceId = Id,
                 };
 
-                vm.FromData(panel);
                 updater.AddOrUpdate(vm);
+                vm.FromData(panel);
             }
         });
 

--- a/src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs
+++ b/src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs
@@ -95,8 +95,21 @@ internal sealed class WorkspaceController : ReactiveObject, IWorkspaceController
             }
             catch (Exception e)
             {
-                _logger.LogError(e, "Exception while restoring Workspace {WorkspaceId}", workspaceData.Id);
-                if (vm.Panels.Count == 0) AddDefaultPanel((WorkspaceViewModel)vm);
+                _logger.LogError(e, "Exception while restoring Workspace `{WorkspaceId}`, resetting to default", workspaceData.Id);
+
+                vm.FromData(new WorkspaceData
+                {
+                    Context = vm.Context,
+                    Id = vm.Id,
+                    Panels = [
+                        new PanelData
+                        {
+                            LogicalBounds = MathUtils.One,
+                            Tabs = [],
+                            SelectedTabId = PanelTabId.DefaultValue,
+                        },
+                    ],
+                });
             }
 
             if (isActiveWorkspace) activeWorkspace = vm;

